src/System.Globalization/tests/NumberStyles/NumberStylesTests.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class NumberStylesTests
+    {
+        [Theory]
+        [InlineData(NumberStyles.AllowCurrencySymbol, 0x00000100)]
+        [InlineData(NumberStyles.AllowDecimalPoint, 0x00000020)]
+        [InlineData(NumberStyles.AllowExponent, 0x00000080)]
+        [InlineData(NumberStyles.AllowHexSpecifier, 0x00000200)]
+        [InlineData(NumberStyles.AllowLeadingSign, 0x00000004)]
+        [InlineData(NumberStyles.AllowLeadingWhite, 0x00000001)]
+        [InlineData(NumberStyles.AllowParentheses, 0x00000010)]
+        [InlineData(NumberStyles.AllowThousands, 0x00000040)]
+        [InlineData(NumberStyles.AllowTrailingSign, 0x00000008)]
+        [InlineData(NumberStyles.AllowTrailingWhite, 0x00000002)]
+        [InlineData(NumberStyles.Any, 0x00000001 | 0x00000002 | 0x00000004 | 0x00000008 | 0x00000010 | 0x00000020 | 0x00000040 | 0x00000080 | 0x00000100)]
+        [InlineData(NumberStyles.Currency, 0x00000001 | 0x00000002 | 0x00000004 | 0x00000008 | 0x00000010 | 0x00000020 | 0x00000040 | 0x00000100)]
+        [InlineData(NumberStyles.Float, 0x00000001 | 0x00000002 | 0x00000004 | 0x00000020 | 0x00000080)]
+        [InlineData(NumberStyles.HexNumber, 0x00000001 | 0x00000002 | 0x00000200)]
+        [InlineData(NumberStyles.Integer, 0x00000001 | 0x00000002 | 0x00000004)]
+        [InlineData(NumberStyles.None, 0x00000000)]
+        [InlineData(NumberStyles.Number, 0x00000001 | 0x00000002 | 0x00000004 | 0x00000008 | 0x00000020 | 0x00000040)]
+        public void Cases(NumberStyles style, int expected)
```

Comment:
Nit: This is a strange name for the test.  I'd expect something more descriptive like "EnumNames_MatchExpectedValues", or something like that.